### PR TITLE
Add entangle defer test and potential fix

### PR DESCRIPTION
**Pull request by joshhanley**
Edit: Please see comment below on the fix.

I have added a failing dusk test for the problem with @entangle.defer I raised in #1874 and also raised in issue #1774 (I realised after I posted).

I hope this helps!

If there is anything else I can do to help, please let me know.

### CODE REVIEW

**Comment by joshhanley** [comment]
I have added a fix for this which ensures that all deferred state changes actually get processed. 

@calebporzio have I done the right thing here? It seems too simple of a fix haha. In your comment above the change I have made, says "This prevents a circular dependancy with the other watcher below", I hope this change doesn't effect this.

Hope this helps! Let me know if I need to change anything.

File: tests/Browser/Alpine/EntangleDefer/Test.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    {
+        $this->browse(function ($browser) {
+            Livewire::visit($browser, Component::class)
+                /**
+                 * Can defer multiple entangle changes until next server request
+                 */
+                ->assertSeeIn('@output.alpine', 'false')
+                ->assertSeeIn('@output.livewire', 'false')
+                ->click('@toggle')
+                ->pause(150)
```
Comment (nuernbergerA) [inline tests/Browser/Alpine/EntangleDefer/Test.php:21]: i dont think you need to pause here

File: tests/Browser/Alpine/EntangleDefer/Test.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+                ->assertSeeIn('@output.alpine', 'true')
+                ->assertSeeIn('@output.livewire', 'false')
+                ->click('@toggle')
+                ->pause(150)
```
Comment (nuernbergerA) [inline tests/Browser/Alpine/EntangleDefer/Test.php:25]: or here

File: tests/Browser/Alpine/EntangleDefer/Test.php (hunk 1 above)
Comment (joshhanley) [reply tests/Browser/Alpine/EntangleDefer/Test.php:21]: Thanks! Agreed. I was looking at some of the other tests when I wrote it, and I think I had a livewire test open at the time and it had pauses in it. I will fix both up.

**Comment by joshhanley** [comment]
Closing this PR as Caleb solved in #1911

**Pull request closed without merging**